src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al

```diff
@@ -0,0 +1,247 @@
+            // Throw an error depending on the reason we couldn't find a date
+            if HideErrorsAndWarnings then
+                exit(false);
+            NoSeriesLine.SetRange("Starting Date");
+            NoSeriesLine.SetRange(Open);
+            if not NoSeriesLine.IsEmpty() then
+                Error(
+                  CannotAssignNewOnDateErr,
+                  NoSeriesCode, UsageDate);
+            Error(
+                CannotAssignNewErr,
+                NoSeriesCode);
+        end;
+
+        // If Date Order is required for this No. Series, make sure the usage date is not before the last date used
+        NoSeries.Get(NoSeriesCode);
+        if NoSeries."Date Order" and (UsageDate < NoSeriesLine."Last Date Used") then begin
+            if HideErrorsAndWarnings then
+                exit(false);
+            Error(
+              CannotAssignNewBeforeDateErr,
+              NoSeries.Code, NoSeriesLine."Last Date Used");
+        end;
+        exit(true);
+    end;
+
+    procedure PeekNextNo(NoSeriesCode: Code[20]; UsageDate: Date): Code[20]
+    var
+        NoSeriesLine: Record "No. Series Line";
+    begin
+        NoSeriesLine."Series Code" := NoSeriesCode;
+        exit(PeekNextNo(NoSeriesLine, UsageDate));
+    end;
+
+    procedure PeekNextNo(var NoSeriesLine: Record "No. Series Line"; UsageDate: Date): Code[20]
+    var
+        NoSeriesSingle: Interface "No. Series - Single";
+    begin
+        if UsageDate = 0D then
+            UsageDate := WorkDate();
+        if not GetNoSeriesLine(NoSeriesLine, NoSeriesLine."Series Code", UsageDate, false) then
+            exit('');
+
+        NoSeriesSingle := GetImplementation(NoSeriesLine);
+
+        exit(NoSeriesSingle.PeekNextNo(NoSeriesLine, UsageDate));
+    end;
+
+    procedure TestAreNoSeriesRelated(DefaultNoSeriesCode: Code[20]; RelatedNoSeriesCode: Code[20])
+    begin
+        if not AreNoSeriesRelated(DefaultNoSeriesCode, RelatedNoSeriesCode) then
+            Error(SeriesNotRelatedErr, DefaultNoSeriesCode, RelatedNoSeriesCode);
+    end;
+
+    procedure AreNoSeriesRelated(DefaultNoSeriesCode: Code[20]; RelatedNoSeriesCode: Code[20]): Boolean
+    var
+        NoSeries: Record "No. Series";
+        NoSeriesRelationship: Record "No. Series Relationship";
+    begin
+        // TODO: call TestIsAutomaticNoSeries instead
+        if not NoSeries.Get(DefaultNoSeriesCode) then
+            exit(false);
+
+        if not NoSeries."Default Nos." then
+            Error(
+              CannotAssignAutomaticallyErr,
+              NoSeries.FieldCaption("Default Nos."), NoSeries.TableCaption(), NoSeries.Code);
+
+        if DefaultNoSeriesCode = RelatedNoSeriesCode then
+            exit(true);
+
+        exit(NoSeriesRelationship.Get(DefaultNoSeriesCode, RelatedNoSeriesCode));
+    end;
+
+
+    procedure IsAutomaticNoSeries(NoSeriesCode: Code[20]): Boolean
+    var
+        NoSeries: Record "No. Series";
+    begin
+        if not NoSeries.Get(NoSeriesCode) then
```

Comment:
When there's a return type, we try to be explicit in what is returned, even if it's the default. To ease readability.
Furthermore we try to move all fault statements to the top for simplicity. It also makes it easier to extend in the future if we need to exit due to some other reason. Instead of adding multiple conditions to the if-statement.

Reply:
unresolving to see if we want to add the SetLoadFields.

Reply:
I have taken a note on this and we will look into it, resolving for now.